pkg/scheduler/service/bookkeeper_test.go

```diff
@@ -85,3 +92,116 @@ func TestBookkeeper(t *testing.T) {
+		{name: "Finish two operations in multiple parallel threads", markOpsDone: true, customFunc: "finishReconciliation", errMessage: "Bookkeeper failed to update cluster", errCount: 49},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			//make sure persistent db is empty
+			err := os.RemoveAll("./test/")
+			require.NoError(t, err)
+			//initialize WaitGroup
+			var wg sync.WaitGroup
+
+			//create mock database connection
+			dbConn := db.NewTestConnection(t)
+			//prepare inventory
+			inventory, err := cluster.NewInventory(dbConn, true, cluster.MetricsCollectorMock{})
+			require.NoError(t, err)
+
+			//add cluster to inventory
+			clusterState, err := inventory.CreateOrUpdate(1, &keb.Cluster{
+				Kubeconfig: "123",
+				KymaConfig: keb.KymaConfig{
+					Components: []keb.Component{
+						{
+							Component:     "dummy",
+							Configuration: nil,
+							Namespace:     "kyma-system",
+						},
+					},
+					Profile: "",
+					Version: "1.2.3",
+				},
+				RuntimeID: uuid.NewString(),
+			})
+			require.NoError(t, err)
+
+			//trigger reconciliation for cluster
+			reconRepo, err := reconciliation.NewPersistedReconciliationRepository(dbConn, true)
+			require.NoError(t, err)
+			reconEntity, err := reconRepo.CreateReconciliation(clusterState, nil)
+			require.NoError(t, err)
+			require.NotEmpty(t, reconEntity.Lock)
+			require.False(t, reconEntity.Finished)
+
+			if tc.markOpsDone {
+				//mark all operations to be done
+				opEntities, err := reconRepo.GetOperations(reconEntity.SchedulingID)
+				require.NoError(t, err)
+				for _, opEntity := range opEntities {
+					err := reconRepo.UpdateOperationState(opEntity.SchedulingID, opEntity.CorrelationID, model.OperationStateDone)
+					require.NoError(t, err)
+				}
+			}
+
+			//initialize logger including error channel
+			errChannel := make(chan error, 100)
+			bookkeeperLogger := zaptest.NewLogger(t, zaptest.WrapOptions(zap.Hooks(func(e zapcore.Entry) error {
+				if strings.Contains(e.Message, tc.errMessage) {
+					errChannel <- errors.New("Update failed")
+				}
+				return nil
+			})))
+
+			//initialize bookkeeper
+			bk := newBookkeeper(
+				newClusterStatusTransition(dbConn, inventory, reconRepo, logger.NewLogger(true)),
+				&BookkeeperConfig{
+					OperationsWatchInterval: 100 * time.Millisecond,
+					OrphanOperationTimeout:  5 * time.Second,
+				},
+				bookkeeperLogger.Sugar(),
+			)
+
+			//setup reconciliation result
+			recons, err := reconRepo.GetReconciliations(nil)
+			require.NoError(t, err)
+			reconResult, err := bk.newReconciliationResult(recons[0])
+			require.NoError(t, err)
+			reconResult.orphanTimeout = 0 * time.Microsecond
+
+			//call markOrphanOperations in parallel threads
+			startAt := time.Now().Add(2 * time.Second)
+			for i := 0; i < 50; i++ {
+				wg.Add(1)
+				go func() {
```

Comment:
Normally for anonymous function, if the logic is simple and just a few lines, we use it for sake of convenience.

In this case, we should create a function with a proper name for it, and use it here, it's self-explanatory and reduces the test body.

Reply:
Lets talk about it ahgain, I would not change it